tests/unit/hooks/test_snapper.py

```diff
@@ -0,0 +1,265 @@
+
+
+@pytest.fixture(scope="function", autouse=True)
+def setup():
+    module._available_configs.cache_clear()
+
+
+@pytest.mark.parametrize("src_dirs_before", ([], ["/foo"], ["/foo", "/bar", "/baz"]))
+def test_unaltered_if_not_configured(src_dirs_before):
+    log_prefix = ''
+    src_dirs_after = module.prepare_source_directories({}, log_prefix, src_dirs_before)
+    assert src_dirs_before == src_dirs_after
+
+
+def test_use_snapshots_include_all():
+    log_prefix = ''
+
+    flexmock(module).should_receive('execute_command_and_capture_output').once().with_args(
+        ["snapper", "--jsonout", "list-configs"]
+    ).and_return(
+        json.dumps(
+            {
+                "configs": [
+                    {
+                        "config": "foo",
+                        "subvolume": "/foo",
+                    },
+                ],
+            }
+        )
+    )
+    flexmock(module).should_receive('execute_command_and_capture_output').once().with_args(
+        ["snapper", "--jsonout", "-c", "foo", "list", "--disable-used-space"]
+    ).and_return(
+        json.dumps(
+            {
+                "foo": [
+                    {
+                        "subvolume": "/foo",
+                        "number": 17,
+                        "default": False,
+                        "active": False,
+                        "type": "single",
+                        "pre-number": None,
+                        "date": "2023-03-02 21:31:44",
+                        "user": "root",
+                        "cleanup": "",
+                        "description": "current",
+                        "userdata": None,
+                    },
+                ]
+            }
+        )
+    )
+    flexmock(module.Path).should_receive("exists").once().and_return(True)
+
+    src_dirs_before = ["/foo"]
+    src_dirs_after = module.prepare_source_directories(
+        {"include": "all"}, log_prefix, src_dirs_before
+    )
+    assert src_dirs_after == ["/foo/.snapshots/17/snapshot"]
+
+
+@pytest.mark.parametrize(
+    "config,src_dirs_in,expected",
+    (
+        (
+            {"include": "all"},
+            ["/foo", "/bar", "/baz"],
+            [
+                "/foo/.snapshots/17/snapshot",
+                "/bar/.snapshots/1337/snapshot",
+                "/baz/.snapshots/13337/snapshot",
+            ],
+        ),
+        (
+            {"include": ["/bar", "/foo"]},
+            ["/foo", "/bar", "/baz"],
+            ["/foo/.snapshots/17/snapshot", "/bar/.snapshots/1337/snapshot", "/baz"],
+        ),
+        (
+            {"include": "all", "exclude": ["/baz"]},
+            ["/foo", "/bar", "/baz"],
+            ["/foo/.snapshots/17/snapshot", "/bar/.snapshots/1337/snapshot", "/baz"],
+        ),
+        (
+            {"include": "all", "exclude": ["/baz"]},
+            ["/foo", "/bar", "/baz"],
+            ["/foo/.snapshots/17/snapshot", "/bar/.snapshots/1337/snapshot", "/baz"],
+        ),
```

Comment:
Unless I'm missing something, this looks like duplicated test parameters. (Same as the last one.)

Reply:
you're right. Sharp eyes :)